src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -298,41 +315,44 @@ impl StaticHeader {
                         match (loc_1, loc_2) {
                             (Some(loc_1), Some(loc_2)) => {
                                 if loc_1 == loc_2 {
-                                    Ok(Some(loc_1))
+                                    Ok(Some(SetupResult::new(loc_1)))
                                 } else if loc_1.initialization_time == loc_2.initialization_time {
                                     // Inexplicable disagreement among static headers
                                     let err_str = "Appeared to be a Stratis device, but signature blocks disagree.";
-                                    Err(StratisError::Engine(ErrorEnum::Invalid, err_str.into()))
+                                    Ok(Some(
+                                        SetupResult::new_with_error(
+                                            loc_1,
+                                            StratisError::Engine(ErrorEnum::Invalid, err_str.into()))))
                                 } else if loc_1.initialization_time > loc_2.initialization_time {
                                     // If the first header block is newer, overwrite second with
                                     // contents of first.
                                     BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
-                                    Ok(Some(loc_1))
+                                    Ok(Some(SetupResult::new(loc_1)))
```

Comment:
_Here_ is where you have to split into two possibilities: if the write succeeded, then ```SetupResult::Ok```, else ```SetupResult::OkWithErr```.

A rule to follow is that ```Err```'s in the old version map to ```Err```s in the new version and the same w/ ```Ok```'s. It's that internal ```SetupResult``` that can now vary based on whether the repair succeeds or not. There are a few more places that need similar adjustment.